src/components/RouteDisplay.tsx

```diff
@@ -75,10 +81,51 @@ const RouteEnd: FC<{
   );
 };
 
-const TransferStep: FC<{ action: TransferAction }> = ({ action }) => {
+const TransferStep: FC<{
+  action: TransferAction;
+  id: string;
+  statusData?: ReturnType<typeof useBroadcastedTxsStatus>["data"];
+}> = ({ action, id, statusData }) => {
   const { data: sourceChain } = useChainByID(action.sourceChain);
   const { data: destinationChain } = useChainByID(action.destinationChain);
 
+  // format: operationType-<operationTypeCount>-<operationIndex>
+  const operationCount = Number(id.split("-")[1]);
+  const transfer = statusData?.transferSequence[operationCount];
+
+  // We can assume that the transfer is successful when the state is TRANSFER_SUCCESS or TRANSFER_RECEIVED
+  const renderState = () => {
+    switch (transfer?.state) {
+      case "TRANSFER_SUCCESS":
+        return (
+          <div className="bg-white rounded">
+            <CheckCircleIcon className="text-green-400 h-6 w-6" />
+          </div>
+        );
+      case "TRANSFER_RECEIVED":
+        return (
+          <div className="bg-white rounded">
+            <CheckCircleIcon className="text-green-400 h-6 w-6" />
+          </div>
+        );
+      case "TRANSFER_FAILURE":
+        return (
+          <div className="bg-white rounded">
+            <XCircleIcon className="text-red-400 h-6 w-6" />
+          </div>
+        );
+      case "TRANSFER_PENDING":
+        return (
+          <div className="bg-white p-1 rounded-full border-2">
+            <Spinner />
+          </div>
+        );
+
+      default:
+        return <div className="w-2 h-2 bg-neutral-200 rounded-full" />;
+    }
+  };
```

Comment:
- either wrap with `useMemo` and render without parentheses (e.g. `{renderState}`)
- or make as a component and wrap with `useCallback` (e.g. `<RenderState />`)